src/mento.ts

```diff
@@ -0,0 +1,322 @@
+    }
+
+    return new Mento(brokerAddr, signerOrProvider)
+  }
+
+  /**
+   * Returns a list of all the pairs that can be traded on Mento
+   * @returns The list of tradeable pairs in the form of [{address, symbol}]
+   */
+  async getTradeablePairs(): Promise<[Asset, Asset][]> {
+    const exchanges = await this.getExchanges()
+    let pairs: [Asset, Asset][] = []
+
+    for (let exchange of exchanges) {
+      const asset0 = exchange.assets[0]
+      const asset1 = exchange.assets[1]
+      const symbols = await Promise.all([
+        getSymbolFromTokenAddress(asset0, this.signerOrProvider),
+        getSymbolFromTokenAddress(asset1, this.signerOrProvider),
+      ])
+      pairs.push([
+        { address: asset0, symbol: symbols[0] },
+        { address: asset1, symbol: symbols[1] },
+      ])
+    }
+
+    return pairs
+  }
+
+  /**
+   * Returns the amount of tokenIn to be sold to buy amountOut of tokenOut
+   * @param tokenIn the token to be sold
+   * @param tokenOut the token to be bought
+   * @param amountOut the amount of tokenOut to be bought
+   * @returns the amount of tokenIn to be sold
+   */
+  async getAmountIn(
+    tokenIn: Address,
+    tokenOut: Address,
+    amountOut: BigNumber
+  ): Promise<BigNumber> {
+    const exchange = await this.getExchangeForTokens(tokenIn, tokenOut)
+    return this.broker.getAmountIn(
+      exchange.providerAddr,
+      exchange.id,
+      tokenIn,
+      tokenOut,
+      amountOut
+    )
+  }
+
+  /**
+   * Returns the amount of tokenOut to be bought by selling amountIn of tokenIn
+   * @param tokenIn the token to be sold
+   * @param tokenOut the token to be bought
+   * @param amountIn the amount of tokenIn to be sold
+   * @returns the amount of tokenOut to be bought
+   */
+  async getAmountOut(
+    tokenIn: Address,
+    tokenOut: Address,
+    amountIn: BigNumber
+  ): Promise<BigNumber> {
+    const exchange = await this.getExchangeForTokens(tokenIn, tokenOut)
+    return this.broker.getAmountOut(
+      exchange.providerAddr,
+      exchange.id,
+      tokenIn,
+      tokenOut,
+      amountIn
+    )
+  }
+
+  /**
+   * Increases the broker's trading allowance for the given token
+   * @param token the token to increase the allowance for
+   * @param amount the amount to increase the allowance by
+   * @returns the populated TransactionRequest object
+   */
+  async increaseTradingAllowance(
+    token: Address,
+    amount: BigNumber
+  ): Promise<providers.TransactionRequest> {
+    if (!Signer.isSigner(this.signerOrProvider)) {
+      throw new Error(
+        'A signer is required to increase the populate the increaseAllowance tx object'
+      )
+    }
+
+    const spender = this.broker.address
+    const tx = await increaseAllowance(
+      token,
+      spender,
+      amount,
+      this.signerOrProvider
+    )
+
+    // The contract call doesn't populate all of the signer fields, so we need an extra call for the signer
+    return this.broker.signer.populateTransaction(tx)
```

Comment:
Feels more natural to use `this.signerOrProvider` because that's what you assert in the beginning of the function.